### PR TITLE
Refactor GUI 

### PR DESCRIPTION
**Pull request by priyanshunayan**
Fixes #

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.5.2 and above.

### Description

This PR does following things:
* Moved all the utility functions to utils folder
* Converted JSX to JS. Because: [No longer Recommended](https://github.com/facebook/create-react-app/issues/87#issuecomment-234627904)
* Made Smaller Components to Print Array, Objects which are reused instead of repeating same code over and over again
* Fixed all previous warnings and errors.
* Made services folder to deal with http requests. This separation of concern helped remove nested promises and made code little bit cleaner.
### Change logs

* Used smaller generic components like PrintObjects, PrintString, PrintID and PrintArray instead of repeating similar type of code in HydraConsole component
* Created utils/utils.js to contain utility functions to be used in anywhere in the app
* Used strict equality and inequality checks to prevent unwanted type coercion
*  Created services for each HTTP endpoint, to follow separation of concern and removed nested promises which could lead to unexpected results.

### CODE REVIEW

File: console-frontend/src/components/hydra-console/HydraConsole.js [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+import PrintObject from "./print-components/PrintObject";
+import PrintArray from "./print-components/PrintArray";
+// utils imports
+import {
+  isArray,
+  isObject,
+  setInLocalStorage,
+  getFromLocalStorage,
+  jsonStringifyReplacer,
+} from "../../utils/utils";
```
Comment (xadahiya) [inline console-frontend/src/components/hydra-console/HydraConsole.js:25]: Can we make this import better somehow? Maybe improve the directory structure a bit?

**Comment by xadahiya** [comment]
Also, can you show some proof that refactoring is not breaking anything?

File: console-frontend/src/components/hydra-console/HydraConsole.js (hunk 1 above)
Comment (priyanshunayan) [reply console-frontend/src/components/hydra-console/HydraConsole.js:25]: I think we can make it little more declarative by putting utils.js in lib folder, or another thing we can do is, in utils folder we can separate files like local-storage-helpers.js (other files similar to this) containing utility functions like getting from local storage and setting values in local storage. Which approach do you think we should move ahead with?

**Comment by priyanshunayan** [comment]
@xadahiya The distinction between js and jsx is no longer useful now. Also, earlier some components were in .js extension and some were in .jsx extension. So, converting to js to jsx provided consistency in the codebase. Also, according to maintainers of create-react-app https://github.com/facebook/create-react-app/issues/87#issuecomment-234627904 .jsx is no longer recommended way to name files.

**Comment by priyanshunayan** [comment]
There are no concrete tests as of now to test things, but I am attaching screenshots of the screen which proves nothing breaks due to refactoring. 
Screenshot 2020-05-30 at 9 43 52 PM

**Comment by chrizandr** [comment]
@xadahiya please review and merge

**Pull request merged**